velox/expression/tests/ExpressionFuzzer.cpp

```diff
@@ -426,12 +426,25 @@ core::TypedExprPtr ExpressionFuzzer::generateArgConstant(const TypePtr& arg) {
       vectorFuzzer_.fuzzConstant(arg, 1));
 }
 
+// Either generates a new column of the required type or if already generated
+// columns of the same type exist then there is a 30% chance that it will
+// re-use one of them.
 core::TypedExprPtr ExpressionFuzzer::generateArgColumn(const TypePtr& arg) {
-  inputRowTypes_.emplace_back(arg);
-  inputRowNames_.emplace_back(fmt::format("c{}", inputRowTypes_.size() - 1));
-
+  auto& listOfCandidateCols = typeToColumnNames_[arg->toString()];
+  bool addNewCol = listOfCandidateCols.empty() || vectorFuzzer_.coinToss(0.3);
+  if (addNewCol) {
+    inputRowTypes_.emplace_back(arg);
+    inputRowNames_.emplace_back(fmt::format("c{}", inputRowTypes_.size() - 1));
+    listOfCandidateCols.push_back(inputRowNames_.back());
+    return std::make_shared<core::FieldAccessTypedExpr>(
+        arg, inputRowNames_.back());
+  }
+  size_t chosenColIndex = listOfCandidateCols.size() == 1
+      ? 0
+      : boost::random::uniform_int_distribution<uint32_t>(
+            0, listOfCandidateCols.size() - 1)(rng_);
```

Comment:
Do we need a special case for size == 1? I saw boost doc says it only requires min <= max.

Reply:
thats just to avoid the cost of creating uniform_int_distribution object and using it. I can remove it for better code readability as our usage here is not performance critical